### PR TITLE
Zip

### PR DESCRIPTION
**Pull request by Snicker7**
Adding the zip command to CRISPRessoShared.py and corresponding logic to CRISPRessoCORE, BATCH, POOLED, and WGS.
If --zip is set to true, place_report_in_output_folder is also set to true.
Zipping files leaves only the zip file removing all files.

### CODE REVIEW

File: CRISPResso2/CRISPRessoBatchCORE.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def propagate_options(cmd, options, params, paramInd):
         if option:
             if option in params:
                 val = params.loc[paramInd, option]
-                if val is None:
+                if val is None or option == "zip":
```
Comment (Colelyman) [inline CRISPResso2/CRISPRessoBatchCORE.py:48]: This is a great way of doing it!

File: CRISPResso2/CRISPRessoShared.py [hunk 2]
```diff
@@ -1372,3 +1598,19 @@ def get_crispresso_footer():
...
 
     return output_line
+
+
+def zip_results(results_folder):
+    path_values = os.path.split(results_folder)
+    output_folder = path_values[0]
+    folder_id = path_values[1]
+    if output_folder == "":
+        cmd_to_zip = 'zip -m -r {0} {1}'.format(
```
Comment (Colelyman) [inline CRISPResso2/CRISPRessoShared.py:1608]: I didn't know about the `-m` option, but it is awesome! Great use of it.

**Review by Colelyman — APPROVED** [review]
This looks good to me! What do you think @kclem?

Also, @Snicker7 when we merge this into master on the Edilytics repo will you open up another PR to merge it into the pinellolab repo? Thanks for adding this feature, I think others will find it very useful.

**Comment by Snicker7** [comment]
Hey @Colelyman,
I just refactored the propagate_crispresso_commands function so there's only a single version instead of Batch having it's own version. This should be simpler to edit in the future. Let me know what you think!

**Review by Colelyman — APPROVED** [review]
I love it! I'm always for getting rid of duplicate code, thanks for noticing this!

**Pull request merged**